File: funcionarios-back-end/src/main/java/com/lucas/henrique/org/funcionarios_crud/Entity/Funcionario.java
package com.lucas.henrique.org.funcionarios_crud.Entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Objects;

@Entity
@Table(name = "funcionario")
@Getter
@Setter
public class Funcionario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String nome;

    @Column
    private String contato_numero;

    @Column
    private String endereco;

    @Column
    private String genero;

    @Column
    private String departamento;

    @Column
    private String skills;



    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        Funcionario that = (Funcionario) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }

}

File: funcionarios-front-end/src/app/model/Funcionario.ts
export interface Funcionario {
    id?:Number | undefined
    nome:string;
    contato_numero:string;
    endereco:string;
    genero:string;
    departamento:string;
    skills:string;
}

export interface FuncionarioResponseId {
    nome:string;
    contato_numero:string;
    endereco:string;
    genero:string;
    departamento:string;
    skills:string;
}

export interface FuncionarioResponse {
    content: Funcionario[];
}